BDArmory/Control/MissileFire.cs

```diff
@@ -6610,9 +6585,12 @@ bool CheckEngagementEnvelope(IBDWeapon weaponCandidate, float distanceToTarget,
                                         float scanSpeed = (rd.Current.locked && rd.Current.lockedTarget.vessel == targetVessel) ? rd.Current.multiLockFOV : rd.Current.directionalFieldOfView / rd.Current.scanRotationSpeed * 2;
                                         if (GpsUpdateMax > 0 && scanSpeed < GpsUpdateMax) GpsUpdateMax = scanSpeed;
                                     }
-                                return true;
                             }
-                            return true;
+                            unguidedWeapon = (ml.TargetingMode == MissileBase.TargetingModes.None || ml.GuidanceMode == MissileBase.GuidanceModes.None) ||
```

Comment:
If any further targeting modes get added to this, then it might make sense to use a switch expression instead, e.g.,
```csharp
unguidedWeapon = ml.GuidanceMode == MissileBase.GuidanceModes.None || ml.TargetingMode switch {
    MissileBase.TargetingModes.None => true,
    MissileBase.TargetingModes.Radar => !_sonarsEnabled && !CurrentMissile.radarLOAL,
    MissileBase.TargetingModes.Inertial => !_sonarsEnabled && !_irstsEnabled
    _ => false
};
```